system-tests/lib/cre/don/jobs/por/por.go

```diff
@@ -210,6 +213,15 @@ func generateDonJobSpecs(
 			} else {
 				jobSpecs[jobDesc] = append(jobSpecs[jobDesc], jobSpec)
 			}
+
+		}
+	}
+
+	// Insert custom jobs, test specific
+	if customJobsFn != nil {
+		jobSpecs, err = customJobsFn(jobSpecs, donWithMetadata)
+		if err != nil {
+			return nil, err
```

Comment:
there's a better way to do it. In general this method was written only for PoR workflow. We shouldn't add anything that's not related here.

This is how I have defined job specs for chainreader: https://github.com/smartcontractkit/chainlink/blob/develop/system-tests/lib/cre/don/jobs/chainreader/chainreader.go

and then this can be merged with jobspecs created for PoR before they are passed to JD for creation.